lms/djangoapps/courseware/views/views.py

```diff
@@ -526,7 +531,11 @@ def register_user_access_warning_messages(request, course_key):
         """
         Register messages to be shown to the user if they have limited access.
         """
-        if request.user.is_anonymous:
+        unenrolled_access_flag = COURSE_ENABLE_UNENROLLED_ACCESS_FLAG.is_enabled(course_key)
+        course = get_course(course_key)
+        allow_anonymous = unenrolled_access_flag and course.course_visibility == COURSE_VISIBILITY_PUBLIC
+
+        if request.user.is_anonymous and not allow_anonymous:
```

Comment:
Nit: It seems like it might be useful to have a helper method that answers `allow_anonymous` for some group of visibility flags (e.g. `allow_anonymous(user, course, course_visibilities)`)?

Performance warning: Calling `get_course` like this is a modulestore call that can potentially make this method much more expensive. If we're already calling `get_course` in some fashion elsewhere in the view where this is invoked, we're fine, because modulestore will cache it. If we're not, it can potentially make the access a lot more expensive. I think we're okay here, but I'm guessing that you know where this is invoked better than I do. Just wanted to make sure you were aware.

Reply:
Thank you for pointing this out. I have modified the function to use `course` passed in from the calling function since they were already fetching it.